File: hikari-query-library/src/main/java/com/joelapp/javabenchmarks/hikariquerylibrary/tables/UserTable.java
package com.joelapp.javabenchmarks.hikariquerylibrary.tables;


import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;
import java.time.LocalDateTime;

public class UserTable extends Table {
    public static void createTable(Connection connection) throws SQLException {
        String sql = """
            CREATE TABLE users (
                id VARCHAR PRIMARY KEY,
                username VARCHAR UNIQUE NOT NULL,
                email VARCHAR UNIQUE NOT NULL,
                created_at TIMESTAMP DEFAULT NOW())
            """;
        try (PreparedStatement statement = connection.prepareStatement(sql)) {
            statement.executeUpdate();
        }
    }

    public static String createID(int userNumber) {
        return createID("USER_", userNumber);
    }

    public static void insertUser(
            Connection connection,
            String userID,
            String username,
            String email
    ) throws SQLException {
        String sql = """
            INSERT INTO users (id, username, email, created_at) VALUES (?, ?, ?, ?)
            """;
        try (PreparedStatement statement = connection.prepareStatement(sql)) {
            statement.setObject(1, userID);
            statement.setString(2, username);
            statement.setString(3, email);
            statement.setObject(4, LocalDateTime.now());
            statement.executeUpdate();
        }
    }
}

File: docker/wrk/scripts/order-items/lib/user-table.ts
import { postgres } from '../../lib/deps.ts';
import { Utils } from '../../lib/utils.ts';

export class UserTable {
  static async createTable(sql: ReturnType<typeof postgres>) {
    await sql`
        CREATE TABLE IF NOT EXISTS users (
          id VARCHAR PRIMARY KEY,
          username VARCHAR UNIQUE NOT NULL,
          email VARCHAR UNIQUE NOT NULL,
          created_at TIMESTAMP DEFAULT NOW()
        )
      `;
  }

  static createID(userNumber: number): string {
    return Utils.createPaddedID('USER_', userNumber);
  }

  static async insertUser(
    sql: ReturnType<typeof postgres>,
    userID: string,
    username: string,
    email: string
  ) {
    await sql`
        INSERT INTO users (id, username, email, created_at)
          VALUES (${userID}, ${username}, ${email}, NOW())
      `;
  }
}
